### PR TITLE
feat: implement new  NoSuchAppOnProviderError component body

### PR DESCRIPTION
**Pull request by themooneer**
### ✅ Checklist

- [ ] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

### 📝 Description

Replace NoSuchAppOnProvider error screen with the new one 
image

### ❓ Context

- **JIRA or GitHub link**: 

https://ledgerhq.atlassian.net/browse/LIVE-14462
---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

File: apps/ledger-live-desktop/src/config/urls.ts [hunk 1]
```diff
@@ -184,6 +184,10 @@ export const urls = {
   howToUpdateNewLedger: "https://support.ledger.com/article/9305992683165-zd",
   genuineCheck: "https://support.ledger.com/article/4404389367057-zd",
   whatAreOrdinals: "https://www.ledger.com/academy/bitcoin-ordinals",
+  ledgerShop:
+    " https://shop.ledger.com?utm_source=live&utm_medium=draw&utm_campaign=ledger_sync_lns_uncompatible&utm_content=to_shop",
```
Comment (LucasWerey) [inline apps/ledger-live-desktop/src/config/urls.ts:188]: extra  unwanted space here

File: apps/ledger-live-desktop/src/renderer/components/DeviceAction/NoSuchAppOnProviderErrorComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+`;
+
+const NoSuchAppOnProviderErrorComponent: React.FC<{
+  error: (Error & { appName: string }) | null | undefined;
+  Icon?: ComponentType<{ color?: string | undefined; size?: number | undefined }>;
+  productName: string;
+}> = ({ error, Icon, productName }) => (
+  <Wrapper id="error-NoSuchAppOnProvider">
+    <ErrorBody
+      Icon={Icon}
```
Comment (LucasWerey) [inline apps/ledger-live-desktop/src/renderer/components/DeviceAction/NoSuchAppOnProviderErrorComponent.tsx:55]: you can do something like that to match with the design :

```
    const theme = useTheme();
    const color = theme.colors.palette.opacityDefault.c05;
    
    top={
      <CircleWrapper color={color} size={72}>
        <Icons.DeleteCircleFill size="L" color="error.c50" />
      </CircleWrapper>
    }
```

File: apps/ledger-live-desktop/src/renderer/components/DeviceAction/NoSuchAppOnProviderErrorComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      }
+      description={
+        <Trans i18nKey="errors.NoSuchAppOnProvider.description" color="neutral.c100">
+          <Text color="neutral.c100" fontWeight="700"></Text>
+        </Trans>
+      }
+    />
+
+    <CtaContainer>
+      <StyledLinkExploreButton
```
Comment (LucasWerey) [inline apps/ledger-live-desktop/src/renderer/components/DeviceAction/NoSuchAppOnProviderErrorComponent.tsx:70]: you can use `<Button variant="main">` from the lib here

File: apps/ledger-live-desktop/src/renderer/components/DeviceAction/NoSuchAppOnProviderErrorComponent.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        label={<Trans i18nKey="errors.NoSuchAppOnProvider.exploreCTA" />}
+        url={urls.ledgerShop}
+      />
+      <StyledLinkLearnMoreButton
```
Comment (LucasWerey) [inline apps/ledger-live-desktop/src/renderer/components/DeviceAction/NoSuchAppOnProviderErrorComponent.tsx:74]: you can use `<Link>` from the lib here

File: apps/ledger-live-desktop/src/renderer/components/DeviceAction/rendering.tsx [hunk 5]
```diff
@@ -784,6 +786,22 @@ export const renderError = ({
...
+        Icon={
+          Icon
+            ? Icon
+            : () => (
+                <Logo info={info} warning={warning}>
+                  <ErrorIcon size={24} error={error} />
+                </Logo>
+              )
+        }
+        error={tmpError}
```
Comment (LucasWerey) [inline apps/ledger-live-desktop/src/renderer/components/DeviceAction/rendering.tsx:801]: ts error and lint error on this one

File: apps/ledger-live-desktop/src/renderer/components/DeviceAction/rendering.tsx [hunk 6]
```diff
@@ -784,6 +786,22 @@ export const renderError = ({
     if (tmpError.message === "User refused") {
       tmpError = new TransactionRefusedOnDevice();
     }
+  } else if(tmpError instanceof NoSuchAppOnProvider) {
+    return (
+      <NoSuchAppOnProviderErrorComponent
+        Icon={
```
Comment (LucasWerey) [inline apps/ledger-live-desktop/src/renderer/components/DeviceAction/rendering.tsx:792]: not needed if we add it directly inside the component or you can set the new design from here

File: apps/ledger-live-desktop/src/renderer/components/DeviceAction/NoSuchAppOnProviderErrorComponent.tsx (hunk 2 above)
Comment (themooneer) [reply apps/ledger-live-desktop/src/renderer/components/DeviceAction/NoSuchAppOnProviderErrorComponent.tsx:55]: nice suggestion, is there any theme guideline that i should follow ? there are a lot of ways to do it , starting from pixel-perfect to using the design system (a lot of pallette and themes here..) Thanks for the review, i will change it

File: apps/ledger-live-desktop/src/renderer/components/DeviceAction/rendering.tsx (hunk 6 above)
Comment (themooneer) [reply apps/ledger-live-desktop/src/renderer/components/DeviceAction/rendering.tsx]: it was already used like this in the last return of this function , that's why i just copied it. 
i've edit it the way you suggest. thanks

File: apps/ledger-live-desktop/src/renderer/components/DeviceAction/NoSuchAppOnProviderErrorComponent.tsx (hunk 2 above)
Comment (LucasWerey) [reply apps/ledger-live-desktop/src/renderer/components/DeviceAction/NoSuchAppOnProviderErrorComponent.tsx]: For the this case color is a assigned like this because CircleWrapper is a styled component. But for other components from the lib you can directly pass color="opacityDefault.c05" for example but they are using the same palette

File: apps/ledger-live-desktop/src/renderer/components/DeviceAction/NoSuchAppOnProviderErrorComponent.tsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
+import React, { ComponentType, useCallback } from "react";
+import { ErrorBody } from "../ErrorBody";
+import { Wrapper } from "./rendering";
+import { Trans } from "react-i18next";
+import { Link, Text } from "@ledgerhq/react-ui";
+import ExternalLinkButton from "../ExternalLinkButton";
```
Comment (LucasWerey) [inline apps/ledger-live-desktop/src/renderer/components/DeviceAction/NoSuchAppOnProviderErrorComponent.tsx:6]: unused import

```suggestion

```

File: apps/ledger-live-desktop/src/renderer/components/DeviceAction/NoSuchAppOnProviderErrorComponent.tsx [hunk 8]
```diff
@@ -0,0 +1,108 @@
+import React, { ComponentType, useCallback } from "react";
```
Comment (LucasWerey) [inline apps/ledger-live-desktop/src/renderer/components/DeviceAction/NoSuchAppOnProviderErrorComponent.tsx:1]: unused import
```suggestion
import React, { useCallback } from "react";
```

File: apps/ledger-live-desktop/src/renderer/components/DeviceAction/rendering.tsx [hunk 9]
```diff
@@ -784,6 +786,13 @@ export const renderError = ({
     if (tmpError.message === "User refused") {
       tmpError = new TransactionRefusedOnDevice();
     }
+  } else if(tmpError instanceof NoSuchAppOnProvider) {
```
Comment (LucasWerey) [inline apps/ledger-live-desktop/src/renderer/components/DeviceAction/rendering.tsx:789]: lint error here
```suggestion
  } else if (tmpError instanceof NoSuchAppOnProvider) {
```

File: apps/ledger-live-desktop/src/renderer/components/DeviceAction/NoSuchAppOnProviderErrorComponent.tsx [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+const NoSuchAppOnProviderErrorComponent: React.FC<{
+  error: Error;
+  productName: string;
+}> = ({ error, productName }) => {
+  const theme = useTheme();
+
+  const handleOnOpenExternalLink = useCallback(
+    (url: string, trackEvent?: string) => () => {
+      openURL(url);
+      if (trackEvent) {
```
Comment (LucasWerey) [inline apps/ledger-live-desktop/src/renderer/components/DeviceAction/NoSuchAppOnProviderErrorComponent.tsx:60]: the condition is not needed the tracking won't work

**Review by LucasWerey — APPROVED** [review]
LGTM

File: apps/ledger-live-desktop/src/config/urls.ts [hunk 11]
```diff
@@ -184,6 +184,10 @@ export const urls = {
   howToUpdateNewLedger: "https://support.ledger.com/article/9305992683165-zd",
   genuineCheck: "https://support.ledger.com/article/4404389367057-zd",
   whatAreOrdinals: "https://www.ledger.com/academy/bitcoin-ordinals",
+  ledgerShop:
```
Comment (valpinkman) [inline apps/ledger-live-desktop/src/config/urls.ts:187]: IMO this should not be part of this PR (but it's not critical to remove it either)

**Review by valpinkman — APPROVED** [review]
